packages/astro/src/runtime/client/dev-toolbar/apps/audit/perf.ts

```diff
@@ -0,0 +1,126 @@
+import type { AuditRuleWithSelector } from './index.js';
+
+// A regular expression to match external URLs
+const externalUrlRe = /^(?:[a-z+]+:)?\/\//i;
+
+export const perf: AuditRuleWithSelector[] = [
+	{
+		code: 'perf-use-image-component',
+		title: 'Use the Image component',
+		message: 'This image could be replaced with the Image component to improve performance.',
+		selector: 'img:not([data-image-component])',
+		async match(element) {
+			const src = element.getAttribute('src');
+			if (!src) return false;
+
+			// Don't match data URIs, they're typically used for specific use-cases that the image component doesn't help with
+			if (src.startsWith('data:')) return false;
+
+			// Ignore images that are smaller than 20KB, most of the time the image component won't really help with these, or they're used for specific use-cases (pixel tracking, etc.)
+			// Ignore this test for remote images for now, fetching them can be very slow and possibly dangerous
+			if (!externalUrlRe.test(src)) {
+				const imageData = await fetch(src).then((response) => response.blob());
+				if (imageData.size < 20 * 1024) return false;
+			}
+
+			return true;
+		},
+	},
+	{
+		code: 'perf-use-loading-lazy',
+		title: 'Use the loading="lazy" attribute',
+		message: (element) =>
+			`This ${element.nodeName} tag is below the fold and could be lazy-loaded to improve performance.`,
+		selector:
+			'img:not([loading]), img[loading="eager"], iframe:not([loading]), iframe[loading="eager"]',
+		match(element) {
+			const htmlElement = element as HTMLImageElement | HTMLIFrameElement;
+			// Ignore elements that are above the fold, they should be loaded eagerly
+			if (htmlElement.offsetTop < window.innerHeight) return false;
+
+			return true;
+		},
+	},
+	{
+		code: 'perf-use-loading-eager',
+		title: 'Use the loading="eager" attribute',
+		message: (element) =>
+			`This ${element.nodeName} tag is above the fold and could be eagerly-loaded to improve performance.`,
+		selector: 'img[loading="lazy"], iframe[loading="lazy"]',
+		match(element) {
+			const htmlElement = element as HTMLImageElement | HTMLIFrameElement;
+			console.log(htmlElement);
+			console.log({ offsetTop: htmlElement.offsetTop, innerHeight: window.innerHeight });
+			// Ignore elements that are below the fold, they should be loaded lazily
+			if (htmlElement.offsetTop > window.innerHeight) return false;
+
+			return true;
+		},
+	},
+	{
+		code: 'perf-use-videos',
+		title: 'Use videos instead of GIFs for large animations',
+		message:
+			'This GIF could be replaced with a video to reduce its file size and improve performance.',
```

Comment:
We should tell the user what "large" means, maybe by explicitly tell the user what's the threshold of this audit